Classes/common/CDTDatastore.m

```diff
@@ -396,6 +396,25 @@ - (BOOL)validateBodyDictionary:(NSDictionary *)body error:(NSError *__autoreleas
             return NO;
         }
     }
+    
+    //now to check if the body contains valid json data
+    if(![NSJSONSerialization isValidJSONObject:body]){
```

Comment:
Should this test go before the "fields contain underscores" check which is above?